### PR TITLE
DOC: Remove duplicate documentation of the PyArray_SimpleNew family of functions

### PR DESCRIPTION
**Pull request by kritisingh1**
Related to #12261

### CODE REVIEW

File: doc/source/user/c-info.how-to-extend.rst [hunk 1]
```diff
@@ -504,50 +504,17 @@ writeable). The syntax is
...
-
-    This function allocates new memory and places it in an ndarray
-    with *nd* dimensions whose shape is determined by the array of
-    at least *nd* items pointed to by *dims*. The memory for the
-    array is uninitialized (unless typenum is :c:data:`NPY_OBJECT` in
-    which case each element in the array is set to NULL). The
-    *typenum* argument allows specification of any of the builtin
-    data-types such as :c:data:`NPY_FLOAT` or :c:data:`NPY_LONG`. The
-    memory for the array can be set to zero if desired using
-    :c:func:`PyArray_FILLWBYTE` (return_object, 0).
```
Comment (mattip) [inline doc/source/user/c-info.how-to-extend.rst]: The last section of this from "The memory is uninitialized (unless ...) should be copied to the other reference in `doc/reference/c-api.array.rst`

File: doc/source/user/c-info.how-to-extend.rst [hunk 2]
```diff
@@ -504,50 +504,17 @@ writeable). The syntax is
...
-    You should ensure that the provided memory is not freed while the
-    returned array is in existence. The easiest way to handle this is
-    if data comes from another reference-counted Python object. The
-    reference count on this object should be increased after the
-    pointer is passed in, and the base member of the returned ndarray
-    should point to the Python object that owns the data. Then, when
-    the ndarray is deallocated, the base-member will be DECREF'd
-    appropriately. If you want the memory to be freed as soon as the
-    ndarray is deallocated then simply set the OWNDATA flag on the
-    returned ndarray.
```
Comment (mattip) [inline doc/source/user/c-info.how-to-extend.rst]: Much of this does not appear in the other reference.

**Comment by kritisingh1** [comment]
@mattip I have updated the pull request. Please review.
Also could you explain the second part of the issue, the cleaning up of DISTUTILS sections? I could not understand it. 
Thanks

**Comment by mattip** [comment]
Looks good now, thanks. 

The second part refers to [this page](http://www.numpy.org/devdocs/reference/distutils_guide.html) which is `doc/DISTUTILS.rst.txt`. It refers to SciPy, not NumPy, and should be a user guide without diving so deeply into detail about the different classes and methods. The classes and methods should be cross-referenced from the [disutils referrence page](http://www.numpy.org/devdocs/reference/distutils.html), there should be a reference from the second page to the first one ("For an overview see ...").

**Pull request merged**